grpc/src/main/java/com/replica/util/RouterConverters.java

```diff
@@ -312,6 +309,16 @@ private static List<String> fetchStreetLegStableIds(Trip.WalkLeg leg) {
                 .collect(toList());
     }
 
+    public static CustomStreetLeg createEmptyCustomStreetLeg(org.locationtech.jts.geom.Point point, Date departureTime,
+                                                             String travelSegmentType, String mode) {
+        // Set departureLocation, instructions, and details params as either "" or null (these values aren't used in
+        // our returned proto objects). Departure and arrival time are set as equal, and distance is set to 0.0
+        Trip.WalkLeg emptyWalkLeg = new Trip.WalkLeg("", departureTime, point, 0.0, null, null, departureTime);
```

Comment:
FYI @rregue 

Right now, I'm setting the departure time equal to the arrival time for these empty walk legs (so they have duration of `0`), and distance as `0.0` meters, with their geometries as a single point and stable edge ID list as an empty list.

Are there any issues we might run into with distances/durations of exactly 0? Is a very small value like `0.1` "safer" here?

I'll take a look around the routing client code + its usages to check, but wanted to note this explicitly too in case any obvious issues exist

Reply:
Not on top of my head. What I am not sure is whether or not the point will cause issues, as in, do we have to have all segments be a segment (line string?)

Reply:
That's a great question too! I checked the code and it turns out the geometry of these legs is actually never used; [the proto objects we generate](https://github.com/replicahq/graphhopper/blob/1d109157528d589175a3524a500d4fcd69d93001/grpc/src/main/java/com/replica/util/RouterConverters.java#L98) for these legs don't actually contain any geometry info, and (to sanity check) [the python code that parses these legs](https://github.com/replicahq/model/blob/58659de6ed3ea644b2b2cd3fe29aa585e326406c/mapservice/api/route.py#L387) doesn't attempt to grab geometry info either (the `TravelSegment` objects that we create from parsing routing results don't store geometry either)

So in theory, we could use anything here, it's just less confusing to use a single point.